### ISSUE TITLE
feat(gcp): Move as many resources as possible to new client

### ISSUE THREAD

**Issue by erezrokah** [cloudquery/cloudquery#6365]
We have a bunch of resources using https://github.com/googleapis/google-api-go-client and not https://github.com/googleapis/google-cloud-go.

Using 2 different clients creates inconsistency in our code, for example only the new client [supports retries](https://discord.com/channels/872925471417962546/1060374442913579019/1060494955832950795).

For some resource we must use the old one (for example https://github.com/cloudquery/cloudquery/pull/5919), but we should try and migrate resources (based on demand).

**Comment by erezrokah** [comment]
If someone wants to contribute see example PR in https://github.com/cloudquery/cloudquery/pull/6390

**Comment by rafabelts** [comment]
Hello, I would like to contribute and work on the migration 😁

**Comment by erezrokah** [comment]
Hi @rafabelts 👋 That's great, please let us know if something is missing from the issue description or comments to get you started.

**Comment by rafabelts** [comment]
Hi👋, just to confirm, I have to migrate from https://github.com/googleapis/google-api-go-client to https://github.com/googleapis/google-cloud-go, right?, but , should some resources like #5919 stay with the old client or should they be moved to the new one?

**Comment by erezrokah** [comment]
> Hi👋, just to confirm, I have to migrate from https://github.com/googleapis/google-api-go-client to https://github.com/googleapis/google-cloud-go, right?, but , should some resources like #5919 stay with the old client or should they be moved to the new one?

Yes, exactly. I recommend first commenting with the list of resources that still use https://github.com/googleapis/google-api-go-client, then we should be able to investigate each one

**Comment by rafabelts** [comment]
Ok ok, well, just one more question, where can I find the resources that are still using the old gcp client?

**Comment by erezrokah** [comment]
Search under this directory https://github.com/cloudquery/cloudquery/tree/b4918ff09f7dae4d20f1beda896d2ec71233716b/plugins/source/gcp/resources/services for resources using `google.golang.org/`. Some are only using `google.golang.org/api/iterator` which is file, but others might be using an old client

**Comment by rafabelts** [comment]
I will check it and then start working, is there any deadline?

**Comment by erezrokah** [comment]
> is there any deadline?

No deadline 😄 Thanks @rafabelts 🙌

**Comment by rafabelts** [comment]
Alright 😄, so if you want you can assign me the issue, I will try to have it ready as soon as possible.

**Comment by erezrokah** [comment]
Going to close this issue as it's not well defined. We can open separate issues per table. @rafabelts if you started working on this please submit any pending work in a PR
